editoast/src/modelsv2/projects.rs

```diff
@@ -0,0 +1,241 @@
+use crate::error::Result;
+use crate::modelsv2::Save;
+use crate::views::pagination::{Paginate, PaginatedResponse};
+use crate::DbPool;
+use actix_web::web::Data;
+use async_trait::async_trait;
+use chrono::{NaiveDateTime, Utc};
+use derivative::Derivative;
+use diesel::sql_types::BigInt;
+use diesel::{sql_query, ExpressionMethods, QueryDsl};
+use diesel_async::{AsyncPgConnection as PgConnection, RunQueryDsl};
+use editoast_derive::ModelV2;
+use serde::{Deserialize, Serialize};
+use utoipa::ToSchema;
+
+use crate::models::List;
+
+crate::schemas! {
+    Ordering,
+    ProjectWithStudies,
+    Project,
+    ProjectRow,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, Derivative, ModelV2, ToSchema)]
```

Comment:
There is a regression in how we manage project images here. At project deletion or update, we check whether the project has an associated image, and if so we delete it.
This behavior is lost with the generated implementation of Delete and Update by ModelV2.

Not sure if in the specific case of documents this should be handled by editoast or by reference counting in the DB (which would allow document sharing). I thought about adding hooks to the model, but I'm still not sure about that, wdyt @flomonster? In the meantime,
I believe we could add a `Document::prune()` function, which would be called by `Project::delete_and_prune_document` and `Project::update_and_prune_document`.

Reply:
We could add a prune function or just leave unused documents. It's not such a big issue.